OpenEdXMobile/src/main/java/org/edx/mobile/view/AccountFragment.kt

```diff
@@ -100,7 +104,7 @@ class AccountFragment : BaseFragment() {
         initHelpFields()
 
         binding.containerPurchases.setVisibility(environment.config.isIAPEnabled)
-        if (!loginPrefs.username.isNullOrBlank()) {
+        if (loginPrefs.isUserLoggedIn) {
```

Comment:
IMO, we should move the delete button functionality behind the user logged-in flag.